src/views/projectExplorer/index.ts

```diff
@@ -166,9 +214,80 @@ export default class ProjectExplorer implements TreeDataProvider<any> {
             }));
           }
           break;
+        case LibraryList.contextValue:
+          const libraryListElement = element as LibraryList;
+          iProject = ProjectManager.get(libraryListElement.workspaceFolder);
+
+          state = await iProject?.getState() as iProjectT;
+          if (state) {
+            const defaultUserLibraries = ibmi?.getConnection().defaultUserLibraries;
+
+            // Retrieve current library
+            let curlib = state.curlib;
+            if (curlib && !curlib.startsWith('&')) {
+              const badLibs = await ibmi?.getContent().validateLibraryList([curlib]);
+              if (badLibs?.includes(curlib)) {
+                curlib = undefined;
+              }
+            } else {
+              curlib = undefined;
+            }
+
+            // Retrieve user libraries
+            let userLibrariesToAdd: string[] = [
+              ...(state.postUsrlibl ? state.postUsrlibl.slice().reverse() : []),
+              ...(defaultUserLibraries ? defaultUserLibraries : []),
+              ...(state.preUsrlibl ? state.preUsrlibl.slice().reverse() : [])
+            ];
+            userLibrariesToAdd = userLibrariesToAdd.filter(lib => !lib.startsWith('&'));
+            const badLibs = await ibmi?.getContent().validateLibraryList(userLibrariesToAdd);
```

Comment:
Perhaps we could combine the two calls to `validateLibraryList` here?

Reply:
Updated this and moved the library list code to iproject.ts